base/filesystem.jl

```diff
@@ -149,6 +149,26 @@ function read(f::File, ::Type{UInt8})
     return ret % UInt8
 end
 
+function read(f::File, ::Type{Char})
+    b0 = read(f, UInt8)
+    n = leading_ones(b0)
+    c = UInt32(b0)
+    if n <= 4
+        while 1 < n
+            p = position(f)
+            b = read(f, UInt8)
+            if b & 0xc0 != 0x80
```

Comment:
Here we should also handle sequences starting with `0xC0`, `0xC1`, `0xE0`, `0xF0` that would be overlong, starting with `0xED` that would be reserved and starting with `0xF4` that would encode too large code point.

Reply:
I'm not sure about that. I think that parsing overlong encodings as a single char is a much friendlier behavior. They occur fairly often in real data, many systems (e.g. [Java](https://en.wikipedia.org/wiki/UTF-8#Modified_UTF-8)) use overlong encodings to avoid embedded nul bytes.